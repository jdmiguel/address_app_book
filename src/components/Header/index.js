import React from 'react';
import PropTypes from 'prop-types';

import { literals } from '../../utils/constants';

const Header = ({ withFinder }) => (
  <header className={!withFinder ? 'basic' : ''}>
    <div className="container">
      <h1>{literals.title}</h1>
    </div>
  </header>
);

Header.propTypes = {
  withFinder: PropTypes.bool,
};

export default Header;
