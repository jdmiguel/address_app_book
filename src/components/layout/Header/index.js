import React from 'react';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';

import Finder from '../../core/Finder';

import { literals } from '../../../utils/constants';

const Header = ({ withFinder, onChangeFinder, isSearching, isMatched }) => (
  <header className={!withFinder ? 'basic' : ''}>
    <div className="container">
      <Link to="/">
        <h1>{literals.logo}</h1>
      </Link>
      {withFinder && (
        <Finder
          onChange={onChangeFinder}
          isSearching={isSearching}
          isMatched={isMatched}
        />
      )}
    </div>
  </header>
);

Header.propTypes = {
  withFinder: PropTypes.bool,
  onChangeFinder: PropTypes.func,
  isSearching: PropTypes.bool,
  isMatched: PropTypes.bool,
};

export default Header;
