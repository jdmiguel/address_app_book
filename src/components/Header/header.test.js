import React from 'react';
import { render } from '@testing-library/react';

import Header from '.';

describe('Component: Header', () => {
  it('should render without Finder', () => {
    const { container } = render(<Header />);

    expect(container.firstChild).toMatchSnapshot();
  });
});
