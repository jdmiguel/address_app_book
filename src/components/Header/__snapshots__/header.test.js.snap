// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component: Header should render with Finder 1`] = `
<header
  class=""
>
  <div
    class="container"
  >
    <h1>
      ADDRESS BOOK
    </h1>
    <input
      autocomplete="nope"
      class=""
      placeholder="Start typing to filter by user..."
    />
  </div>
  <div />
</header>
`;

exports[`Component: Header should render without Finder 1`] = `
<header
  class="basic"
>
  <div
    class="container"
  >
    <h1>
      ADDRESS BOOK
    </h1>
  </div>
  <div />
</header>
`;
